plugins/woocommerce/src/Internal/Admin/Agentic/AgenticWebhookPayloadBuilder.php

```
@@ -0,0 +1,230 @@
+<?php
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Internal\Admin\Agentic;
+
+use Automattic\WooCommerce\Enums\OrderStatus;
+use Automattic\WooCommerce\StoreApi\Routes\V1\Agentic\Enums\Specs\OrderStatus as ACPOrderStatus;
+use WC_Order;
+use WC_Order_Refund;
+
+/**
+ * AgenticWebhookPayloadBuilder class
+ *
+ * Builds webhook payloads for the Agentic Commerce Protocol following
+ * the specification for order lifecycle events.
+ *
+ * @since 10.3.0
+ */
+class AgenticWebhookPayloadBuilder {
+	/**
+	 * Build the webhook payload for an order event.
+	 *
+	 * @param string   $event Event type ('order_create' or 'order_update').
+	 * @param WC_Order $order Order object.
+	 * @return array Webhook payload.
+	 */
+	public function build_payload( $event, $order ) {
+		return array(
+			'type' => $event,
+			'data' => $this->build_order_data( $order ),
+		);
+	}
+
+	/**
+	 * Build the order data for the webhook payload.
+	 *
+	 * @param WC_Order $order Order object.
+	 * @return array Order data.
+	 */
+	private function build_order_data( $order ) {
+		return array(
+			'type'                => 'order',
+			'checkout_session_id' => $this->get_checkout_session_id( $order ),
+			'permalink_url'       => $this->get_order_permalink( $order ),
+			'status'              => $this->map_order_status( $order->get_status() ),
+			'refunds'             => $this->build_refunds_data( $order ),
+		);
+	}
+
+	/**
+	 * Get the checkout session ID from the order.
+	 *
+	 * @param WC_Order $order Order object.
+	 * @return string Checkout session ID.
+	 */
+	private function get_checkout_session_id( $order ) {
+		$session_id = $order->get_meta( '_agentic_checkout_session_id' );
+		return ! empty( $session_id ) ? $session_id : 'checkout_session_' . $order->get_id();
```

Is there a scenario where an order might not have the session ID in meta? I'd rather throw an exception and terminate the process here.

FWIW the meta is already required in `AgenticWebhookManager->should_trigger_webhook()` so I don't see a reason to support scenarios with missing `_agentic_checkout_session_id`.